portfolio/src/main/webapp/reusehtml.js

```diff
@@ -0,0 +1,48 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * Adds HTML content to elements with the include-html attribute
+ */
+function includeHtml() {
+  const elementsWithIncludeAttr = document.querySelectorAll("[include-html]");
+
+  for (const element of elementsWithIncludeAttr) {
+    // Get the file name
+    const path = element.getAttribute("include-html");
+    
+    // Check path is not an empty string
+    if (path) {
+      // Make an HTTP Request for the HTML file
+      const xmlHttp = new XMLHttpRequest();
+      xmlHttp.onreadystatechange = function() {
+        if (xmlHttp.readyState === 4) {
```

Comment:
I'm not sure how many states there can be, but perhaps a fallback here as well. Are there error states, what if it hangs, is there a timeout on the request?

Reply:
There are 5, from 0 to 4. They monitor the progress, with 4 indicating that the operation is complete. I include a loading message while the state is not 4.
I also added a timeout of 5 seconds, when it is reached for the first time it calls a function to try the request again, if it fails again it shows a timeout message.
To handle errors, xmlHttp.onerror was defined.